src/core/qgsproject.h

```diff
@@ -580,6 +581,21 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     QgsProjectViewSettings *viewSettings();
 
+    /**
+     * Returns the project's display settings, which settings and properties relating
+     * to how a QgsProject should display values such as map coordinates and bearings.
+     * \note not available in Python bindings
+     * \since QGIS 3.12
+     */
+    const QgsProjectDisplaySettings *displaySettings() const SIP_SKIP;
+
+    /**
+     * Returns the project's display settings, which settings and properties relating
+     * to how a QgsProject should display values such as map coordinates and bearings.
+     * \since QGIS 3.12
+     */
+    QgsProjectDisplaySettings *displaySettings();
```

Comment:
Any philosophical reason why you've decided not to add a setter here?

Reply:
Yes -- partly because it's a QObject (and there may be connections setup to the existing settings object, so deleting it would break those), and also because it's following the pattern used elsewhere in QgsProject (e.g. layoutManager, annotationManager, etc)

Reply:
@nyalldawson , all good then. Hit that merge button! :)

Reply:
I guess this discussion arose already in the past, but we're mixing a lot the 2 things in QGIS (modifiable QObject vs getter/setter of non-QObject based config).
ping @m-kuhn 
Wouldn't it be nice to come to a consensus and stick to it?
Or are there cases where one is better than the other?

Reply:
I think in this case:

1. We definitely want QgsProjectDisplaySettings to be a QObject on its own -- otherwise the signals from it would need to be moved to the already very complex QgsProject class. (The intention with classes like this one and QgsProjectViewSettings was to start breaking down the QgsProject behemoth into more maintainable smaller classes to avoid QgsProject becoming a dumping ground for everything.)

2. If QgsProjectDisplaySettings is a QObject, then we expect things to connect to its signals (in this case, QgisApp connects to it). If we make a setter for this and require api users to create a new QgsProjectDisplaySettings instance and set it on the project, any existing connections would be lost. We'd then need to add a new signal "projectDisplaySettingsChanged" to QgsProject. Instead of having granular signals in QgsProjectDisplaySettings, clients could only connect to this one all-encompassing signal and respond to it even when the actual setting they care about hasn't been the one which was changed.

Reply:
This makes a lot of sense, and I recall thinking it was the way to go.
Trying to recall discussions on this, the snapping config came to my mind:
https://github.com/qgis/QGIS/pull/3617#issuecomment-254414285

My point here is really not towards your work, but rather to decide which is the way to go for the standard cases. And to me the snapping config is a really good comparison to this.